File: server/serverpackets/S_CharAmount.py
# -*- coding: utf-8 -*-

from Config import Config
from server.Account import Account
from server.codes import Opcodes
from ServerBasePacket import ServerBasePacket

class S_CharAmount(ServerBasePacket):
    def __init__(self, value, client):
        ServerBasePacket.__init__(self)
        account = Account.load(client._account._name)
        maxAmount = Config.getint('altsettings', 'DefaultCharacterSlot') + account._characterSlot
        self.writeC(Opcodes.S_OPCODE_CHARAMOUNT)
        self.writeC(value)
        self.writeC(maxAmount)

    def getContent(self):
        return self.getBytes()

File: ServerPackets/S_CharAmount.cc
//
// Created by 范炜东 on 2019/5/9.
//

#include "Config.h"
#include "Opcodes.h"
#include "S_CharAmount.h"

S_CharAmount::S_CharAmount(int count, int slot) : ServerPacket() {
  writeC(Opcodes::S_OPCODE_CHARAMOUNT);
  writeC(count);
  writeC(Config::DEFAULT_CHARACTER_SLOT + slot);
}